src/Build/BackEnd/BuildManager/BuildSubmission.cs

```diff
@@ -156,12 +156,12 @@ internal void CompleteResults(BuildResult result)
             // this one.)
             ErrorUtilities.VerifyThrow(result.ConfigurationId == BuildRequest.ConfigurationId, "BuildResult doesn't match BuildRequest configuration");
 
-            if (BuildResult == null)
+            if (BuildResult is null)
```

Comment:
It's still not clear to me why this needs to change. What's the case where 

1. `CompleteResults` is called with a passing result, and then
2. `CompleteResults` is called with a failing errored-out result

? I get the "there may be multiple requests for one configuration" but not "they have different results".

Reply:
It's possible it doesn't. I noticed this bit from just looking at the code, noticing there can be multiple requests for one configuration, and assuming they might disagree. Then I looked at it in a debugger and tweaked variables. I don't actually know for sure if there's a way a user could get into that state, but I saw this as a "I don't think it would hurt, but there's probably some way it will help for some weird edge case" case. If you feel that's too risky or there is no way for that to happen, I can revert it. I was testing with one request, for what it's worth.

Reply:
Since neither of us understands it deeply I'd prefer to revert.